### PR TITLE
Filter cancer variants by old local germline and old local somatic observations

### LINKED ISSUE #4349: somatic view variantS archive variants filtering
Allow filtering also on cancer-germline and somatic variants.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4349

Cancer variant filter before:

image

After:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Deploy on stage and do some variantS filtering

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: tests/adapter/mongo/test_query.py [hunk 1]
```diff
@@ -655,27 +655,33 @@ def test_build_swegen_sv(adapter):
...
 
-def test_build_local_obs_sv(adapter):
-    """Test building query. Given query parameters from interface, check that
-    a correct db query is made. This tests local observation counts for SVs.
-    """
+def test_build_local_obs(adapter):
+    """Test building a query containing local archived observations (RD and cancer variants."""
 
     # GIVEN query parameters for local observations
+    LOCAL_OLD_OBS_FIELDS = [
```
Comment (dnil) [inline tests/adapter/mongo/test_query.py:662]: 🥇

File: scout/server/blueprints/variants/templates/variants/utils.html [hunk 2]
```diff
@@ -289,8 +310,7 @@
           {{ form.gnomad_frequency(class="form-control") }}
         </div>
         <div class="col-4">
-          {{ form.local_obs.label(class="control-label") }}
-          {{ form.local_obs(class="form-control") }}
+          {{ archived_observations_filter(form) }}
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/utils.html:313]: 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - Display paging and number of HPO terms available in the database on Phenotypes page
 - On case page, typeahead hints when searching for a disease using substrings containing source ("OMIM:", "ORPHA:")
 - Button to monitor the status of submissions on ClinVar Submissions page
+- Option to filter cancer variants by number of observations in somatic and germiline archived database
```
Comment (dnil) [inline CHANGELOG.md:15]:
```suggestion
- Option to filter cancer variants by number of observations in somatic and germline archived database
```

File: scout/server/blueprints/variants/forms.py [hunk 4]
```diff
@@ -166,6 +166,12 @@ class CancerFiltersForm(VariantFiltersForm):
     clinvar_tag = BooleanField("ClinVar hits")
     cosmic_tag = BooleanField("Cosmic hits")
     mvl_tag = BooleanField("Managed Variants hits")
+    local_obs_cancer_somatic_old = IntegerField(
+        "Local somatic obs. (archive)", validators=[validators.Optional()]
+    )
+    local_obs_cancer_germline_old = IntegerField(
+        "Local germile obs. (archive)", validators=[validators.Optional()]
```
Comment (dnil) [inline scout/server/blueprints/variants/forms.py:173]:
```suggestion
        "Local germline obs. (archive)", validators=[validators.Optional()]
```

File: scout/server/blueprints/variants/forms.py [hunk 5]
```diff
@@ -197,6 +203,13 @@ class SvFiltersForm(VariantFiltersForm):
     swegen = IntegerField("SweGen obs")
     clinical_filter = SubmitField(label="Clinical filter")
 
+    local_obs_cancer_somatic_old = IntegerField(
+        "Local somatic obs. (archive)", validators=[validators.Optional()]
+    )
+    local_obs_cancer_germline_old = IntegerField(
+        "Local germile obs. (archive)", validators=[validators.Optional()]
```
Comment (dnil) [inline scout/server/blueprints/variants/forms.py:210]:
```suggestion
        "Local germline obs. (archive)", validators=[validators.Optional()]
```

**Review by dnil — APPROVED** [review]
Great! Note that the spellcheck isn't super with genetic terms; germline e.g. - at least it wasn't gremlins! 😁

**Comment by northwestwitch** [comment]
> Great! Note that the spellcheck isn't super with genetic terms; germline e.g. - at least it wasn't gremlins! 😁

Gemiline, right! 😆 Thanks for the review!

**Pull request merged**